### PR TITLE
add: cart management

### PR DESCRIPTION
**Pull request by thegeorgenikhil**
Added Cart Management to the app, now users can:
- add items to the cart
- remove items from the cart
- increment or decrement the quantity of the items in the cart

Extra features:
- added a test user credential button for testing out the app
- calculates the price of the total items in the cart
- logs the user in if the token is present in the local storage(***NOTE:**: currently works only for the test user as it is the only user available in the DB, so only the test user data will be persisting in between reloads*)

### CODE REVIEW

File: src/components/Navbar/Navbar.css [hunk 1]
```diff
@@ -69,6 +69,20 @@
...
+}
+
+.icon-badge{
+  position: absolute;
+  top: -10px;
+  right: -10px;
+  background-color: var(--secondary-color);
+  padding: 2px 6px;
+  border-radius: 100%;
+  font-size: 10px;
```
Comment (KamiyaGaikwad) [inline src/components/Navbar/Navbar.css:87]: Try to avoid using px

**Review by KamiyaGaikwad — APPROVED** [review]
Good to merge
1. Good file structure
2. Code indentations are proper
3. Great u added test credentials and really awesome app UI. Great Job

File: src/components/Navbar/Navbar.css (hunk 1 above)
Comment (thegeorgenikhil) [reply src/components/Navbar/Navbar.css:87]: I have used px as the value is less than 10 and I want that value to remain constant throughout in all screen sizes and not change like as in the case of rems.

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -40,20 +45,25 @@ const Navbar = () => {
           </div>
         </li>
         <li className="nav-item">
-          <Link className="nav-item-link" to="/cart">
-            <AiOutlineShoppingCart fontSize={"1.4rem"} />
-          </Link>
+          <div className="icon-badge-container">
+            <Link className="nav-item-link" to="/cart">
+              <AiOutlineShoppingCart fontSize={"1.4rem"} />
```
Comment (surajmore98) [inline src/components/Navbar/Navbar.jsx]: you can create class for fontsize 1.4rem and use that.

File: src/components/SignIn/SignIn.jsx [hunk 3]
```diff
@@ -75,9 +88,18 @@ const SignIn = () => {
...
+            setFormData({ email: "test@gmail.com", password: "test123" });
+          }}
+          className={`btn form-signup-btn`}
+        >
+          Use Test Credentials
+        </button>
       </form>
       <p className="text-center">
-        Don't have an account? Sign up!
+        Don't have an account? Sign up!{" "}
```
Comment (surajmore98) [inline src/components/SignIn/SignIn.jsx]: you can remove this.

File: src/pages/Home.jsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
-import React from "react";
+import React,{useEffect} from "react";
 import CategoriesSection from "../components/CategoriesSection/CategoriesSection";
 import Footer from "../components/Footer/Footer";
 import HomeCarousel from "../components/HomeCarousel/HomeCarousel";
 import Navbar from "../components/Navbar/Navbar";
 
 const Home = () => {
+  useEffect(() => {
+    const token = localStorage.getItem("")
```
Comment (surajmore98) [inline src/pages/Home.jsx]: not sure what is use of this.

File: src/reducers/userInfoReducer.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+          ...state.cartItems,
+          { ...action.payload.product, quantity: 1 },
+        ],
+      };
+    case "DECREMENT_CART_PRODUCT":
+      const indexOfProductInCart = state.cartItems.findIndex(
+        (item) => item._id === action.payload.product._id
+      );
+      console.log(indexOfProductInCart);
+      console.log(state.cartItems);
```
Comment (surajmore98) [inline src/reducers/userInfoReducer.js]: you can remove this logs if not needed.

**Review by surajmore98 — APPROVED** [review]
liked the test login functionality easy to see preview.
well structured code.
with above suggestions it is good to merge.

File: src/components/SignIn/SignIn.jsx (hunk 3 above)
Comment (thegeorgenikhil) [reply src/components/SignIn/SignIn.jsx]: It was added by prettier, fixed it!

File: src/pages/Home.jsx (hunk 4 above)
Comment (thegeorgenikhil) [reply src/pages/Home.jsx]: forgot to remove that, thank you.

File: src/reducers/userInfoReducer.js (hunk 5 above)
Comment (thegeorgenikhil) [reply src/reducers/userInfoReducer.js]: done

**Comment by thegeorgenikhil** [comment]
> liked the test login functionality easy to see preview. well structured code. with above suggestions it is good to merge.

Thank you, fixed all the issues.

**Comment by thegeorgenikhil** [comment]
> Good to merge
> 
> 1. Good file structure
> 2. Code indentations are proper
> 3. Great u added test credentials and really awesome app UI. Great Job

Thank you @KamiyaGaikwad

**Pull request merged**